### ISSUE TITLE
.inputField(): More input control (erase begining of input, erase end of input, move to next/previous word)

### ISSUE THREAD

**Issue by jwarkentin** [cronvel/terminal-kit#21]
I'm trying to implement the `CTRL_U` and `CTRL_K` functionality that erases from the cursor to the beginning of the line and from the cursor to the end of the line, respectively. Unfortunately I can't use the actual control codes for this because the control codes are only erasing the text on the screen, not changing the actual text in the buffer. Also, calling `term.eraseLineBefore()` (which sends the control code) will erase the whole line including the prompt text for the input field.

My next approach was to try to get the total input string length from the input field, along with the cursor position and call `backDelete()` or `delete()` for each character until I've erased to the beginning or end of the input string. However, the position returned from the call to `inputField.getPosition()` is just the cursor position in the terminal, not the position in the input string. And since there is prompt text before it I would have to do a lot more work to subtract the length of the prompt text and figure out if the command is wrapping onto multiple lines and hope I didn't miss something, just so I could hopefully get the right cursor position in the input string.

I think ideally there would be a simple way to set the input value, not just get it. And whether that happens or not it would really be helpful if there was a way to get and set the cursor position in the input string rather than working with the raw terminal coordinates. What are your thoughts? Do you think there is a better way to solve this problem? Or would this be an easy thing to implement?

**Comment by cronvel** [comment]
BTW you should not edit the screen directly because it would do it behind the back of the inputField.

If I understand correctly, this is trivial to implement, I will look into it tomorrow.

**Comment by jwarkentin** [comment]
Another thing I'm thinking about where this would be useful is in implementing the standard terminal ability to hold CTRL and press the left and right arrow keys to move the cursor by words and logical segments. If there were just a way to get and set the cursor offset (single number, not coordinates) in the input relative to the beginning of the text then all of this would be easy to implement. The trick is just making sure that the offset takes into account multi-line inputs.

**Comment by cronvel** [comment]
@jwarkentin Ctrl-left, Ctrl-right, Ctrl-U and Ctrl-K are now implemented as of v1.2.0.

**Comment by samizdatco** [comment]
As far as I can tell, the current implementation of the Ctrl-U and Ctrl-K behavior updates the cursor position and the inputField's value, but it doesn't actually erase any characters from the terminal. Is this a bug or is the screen-updating supposed to happen outside of the inputField's redraw() routine?

**Comment by cronvel** [comment]
@samizdatco It works on all terminal that I tested. What is your terminal/OS?

**Comment by samizdatco** [comment]
I'm using Terminal.app on Mac OS 10.11 (though note that the same behavior is seen in iTerm2 as well).

Here's my test script:
```js
var term = require( 'terminal-kit' ).terminal
term('>')
term.inputField(function(err, input){
    term.processExit(0)
})
```

And here's what I see after entering some text and then typing Ctrl-U:

![termkit-crtl-u](https://cloud.githubusercontent.com/assets/469523/26737593/bced5c30-4798-11e7-9434-4aa4043e78f1.png)

**Comment by cronvel** [comment]
Thanks for reporting, indeed there was redraw issue depending on some options (or absence of options). It should be fixed now! ;)

**Comment by samizdatco** [comment]
Yes! It's working great now. Thanks so much!

**Comment by samizdatco** [comment]
I also added a couple of other deletion keybindings that are common to most readline-powered command line tools [here](https://github.com/samizdatco/terminal-kit/commit/1c2f222e24bdb0d0011f555da73a52e667cee205):
  Ctrl-W: delete previous word
  Alt-D: delete next word

Do these seem more or less correct to you?

**Comment by cronvel** [comment]
This is cherry-picked, and it landed on v1.9.0, thanks! ;)
